qiskit/visualization/gate_map.py

```diff
@@ -16,11 +16,16 @@
 from typing import List
 
 import numpy as np
-import rustworkx as rx
+from rustworkx.visualization import graphviz_draw
+
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib import gridspec, ticker
```

Comment:
These imports will need to be done at run time, because matplotlib isn't guaranteed to be avaliable when this module is imported. So we'll need these imports inside the functions where they're used to avoid a potential import error for users that don't have matplotlib installed.

I actually tested doing this locally in a windows vm with python 3.8 and it also fixed the segfault errors that windows CI is failing on. @jakelishman and I were discussing this offline and think it is likely a side effect of a stack limit being hit by having extra modules in the import path somehow (the exact path isn't clear). But anyway moving these imports to be done at runtime in the functions should fix the CI failures too.

Reply:
I went ahead and fixed this in: https://github.com/Qiskit/qiskit-terra/pull/10208/commits/d719620000dc1d3e4114fa876828865a7485b7b5